internal/envoy/v3/route.go

```diff
@@ -151,16 +164,25 @@ func routeAuthzContext(settings map[string]string) *any.Any {
 
 // RouteMatch creates a *envoy_route_v3.RouteMatch for the supplied *dag.Route.
 func RouteMatch(route *dag.Route) *envoy_route_v3.RouteMatch {
-	switch c := route.PathMatchCondition.(type) {
+	routeMatch := PathRouteMatch(route.PathMatchCondition)
+
+	routeMatch.Headers = headerMatcher(route.HeaderMatchConditions)
+	routeMatch.QueryParameters = queryParamMatcher(route.QueryParamMatchConditions)
+
+	return routeMatch
+}
+
+// PathRouteMatch creates a *envoy_route_v3.RouteMatch with *only* a PathSpecifier
+// populated.
+func PathRouteMatch(pathMatchCondition dag.MatchCondition) *envoy_route_v3.RouteMatch {
+	switch c := pathMatchCondition.(type) {
 	case *dag.RegexMatchCondition:
 		return &envoy_route_v3.RouteMatch{
 			PathSpecifier: &envoy_route_v3.RouteMatch_SafeRegex{
 				// Add an anchor since we at the very least have a / as a string literal prefix.
 				// Reduces regex program size so Envoy doesn't reject long prefix matches.
 				SafeRegex: SafeRegexMatch("^" + c.Regex),
 			},
-			Headers:         headerMatcher(route.HeaderMatchConditions),
```

Comment:
nice!

Reply:
:+1: turns out this wasn't actually necessary since the rules moved to inline in the routes, but it's a good cleanup anyway so will leave it in place.